github-runners/terraform/base/main.tf

```diff
@@ -0,0 +1,440 @@
+  #     # configure runner
+  #     "./config.sh --url https://github.com/subspace --token ${var.GH_TOKEN} --unattended --name linux_arm64 --labels 'self-hosted,Linux,ARM64' --work _work --runasservice",
+  #     "sudo ./svc.sh install",
+  #     "sudo ./svc.sh start",
+  #     "sudo ./svc.sh status",
+  #     # install monitoring
+  #     "sudo wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --non-interactive --nightly-channel --claim-token ${var.netdata_token} --claim-url https://app.netdata.cloud",
+
+  #   ]
+  # }
+
+  # provisioner "file" {
+  #   source      = "./scripts/cleanup.sh"
+  #   destination = "/home/ubuntu/cleanup.sh"
+  # }
+
+  # # Setting up the ssh connection to install the runner server
+  # connection {
+  #   type        = "ssh"
+  #   host        = element(self.*.public_ip, count.index)
+  #   user        = "ubuntu"
+  #   private_key = file("${var.public_key_path}")
+  #   timeout     = "15m"
+  # }
+
+}
+
+resource "aws_instance" "mac_x86_runner" {
+  count             = length(var.public_subnet_cidrs)
+  ami               = data.aws_ami.mac_x86.image_id
+  instance_type     = element(var.instance_type_mac, 0)
+  subnet_id         = element(aws_subnet.public_subnets.*.id, count.index)
+  availability_zone = element(var.azs, count.index)
+  tenancy           = "dedicated"
+  # Security Group
+  vpc_security_group_ids = ["${aws_security_group.allow_runner.id}"]
+  # the Public SSH key
+  key_name                    = var.aws_key_name
+  associate_public_ip_address = true
+
+  tags = {
+    name       = "gh-macos-x86-runner"
+    role       = "runner"
+    os_name    = "macos"
+    os_version = "12"
+    os_name    = "Monterey"
+    arch       = "x86"
+  }
+
+  depends_on = [
+    aws_subnet.public_subnets,
+    aws_nat_gateway.nat_gateway,
+    aws_internet_gateway.gw
+  ]
+
+  lifecycle {
+    # ignore_changes = [
+    #   tags, ami
+    # ]
+    create_before_destroy = true
+
+    //ignore_changes = all
+  }
+
+  user_data = <<-EOF
+  #!/bin/bash
+  sudo apt update -y
+  sudo apt upgrade -y
+  sudo apt install git curl jq wget gnupg openssl make build-essential jq net-tools -y
+  sudo curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain nightly -y
+  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+  mkdir actions-runner && cd actions-runner
+  curl -o actions-runner-osx-x64-2.303.0.tar.gz -L https://github.com/actions/runner/releases/download/v2.303.0/actions-runner-osx-x64-2.303.0.tar.gz
+  echo '8bd595568ceee5eb25576972bc8075b47c149b3fac7eb7873deed67944b45739  actions-runner-osx-x64-2.303.0.tar.gz' | shasum -a 256 -c
+  tar xzf ./actions-runner-osx-x64-2.303.0.tar.gz
+  ./config.sh --url https://github.com/subspace --token ${var.GH_TOKEN} --unattended --name mac_x86_64 --labels 'self-hosted,MacOS,X64' --work _work --runasservice
```

Comment:
In one place it is `x86`, in other `amd64`, here it is `X64`. Can we call them consistently or they have to be called this way?